### PR TITLE
Add start/end timestamps to test results

### PR DESCRIPTION
**Pull request by bryevdv**
This PR adds start and end timestams to the test output, in case it is needed for detailed observation of the test plan exectution.  

image

### CODE REVIEW

File: legate/tester/test_system.py [hunk 1]
```diff
@@ -43,11 +43,14 @@ class ProcessResult:
...
-    #  User-friendly test file path to use in reported output
+    #: User-friendly test file path to use in reported output
     test_file: Path
 
-    #: The time the process took to execute in seconds
-    time: timedelta | None = None
+    #: The time the process started
+    start: datetime | None = None
+
+    #: The time the process started
```
Comment (manopapad) [inline legate/tester/test_system.py:52]:
```suggestion
    #: The time the process ended
```

File: legate/tester/stages/util.py [hunk 2]
```diff
@@ -123,11 +123,15 @@ def log_proc(
...
-        if proc.time is not None
-        else ""
-    )
+
+    if proc.time is None or proc.start is None or proc.end is None:
+        duration = ""
+    else:
+        time = yellow(f"{proc.time.total_seconds():0.2f}s")
+        start = f"{proc.start.strftime('%H:%M:%S.%f')[:-4]}"
+        end = f"{proc.end.strftime('%H:%M:%S.%f')[:-4]}"
```
Comment (manopapad) [inline legate/tester/stages/util.py:132]:
```suggestion
        time = f"{proc.time.total_seconds():0.2f}s"
        start = proc.start.strftime("%H:%M:%S.%f")[:-4]
        end = proc.end.strftime("%H:%M:%S.%f")[:-4]
```

**Review by manopapad — APPROVED** [review]
Nits

File: legate/tester/stages/util.py (hunk 2 above)
Comment (bryevdv) [reply legate/tester/stages/util.py:132]: I mean I guess you could not-use f-strings who knew that was possible :rofl:

**Pull request merged**